File: support_tools/swigwin-3.0.12/Examples/test-suite/go/typedef_inherit_runme.go
package main

import "./typedef_inherit"

func main() {
	a := typedef_inherit.NewFoo()
	b := typedef_inherit.NewBar()

	x := typedef_inherit.Do_blah(a)
	if x != "Foo::blah" {
		panic(x)
	}

	x = typedef_inherit.Do_blah(b)
	if x != "Bar::blah" {
		panic(x)
	}

	c := typedef_inherit.NewSpam()
	d := typedef_inherit.NewGrok()

	x = typedef_inherit.Do_blah2(c)
	if x != "Spam::blah" {
		panic(x)
	}

	x = typedef_inherit.Do_blah2(d)
	if x != "Grok::blah" {
		panic(x)
	}
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/typedef_inherit_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'typedef_inherit'

a = Typedef_inherit::Foo.new
b = Typedef_inherit::Bar.new

x = Typedef_inherit.do_blah(a)
if x != "Foo::blah"
  puts "Whoa! Bad return #{x}"
end

x = Typedef_inherit.do_blah(b)
if x != "Bar::blah"
  puts "Whoa! Bad return #{x}"
end

c = Typedef_inherit::Spam.new
d = Typedef_inherit::Grok.new

x = Typedef_inherit.do_blah2(c)
if x != "Spam::blah"
  puts "Whoa! Bad return #{x}"
end

x = Typedef_inherit.do_blah2(d)
if x != "Grok::blah"
  puts "Whoa! Bad return #{x}"
end
